app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+class CommentsController < ApplicationController
+  before_action :set_comment, only: [:destroy]
+
+  def comments_by_post
+    respond_to do |format|
+      if @post = Post.find_by(id: params[:post_id])
+        @comments = @post.comments.paginate(page: params[:comments_page], per_page: 3)
+        format.html { redirect_to @post }
+        format.js
+      else
+        format.html { redirect_back fallback_location: root_path, danger: 'Something went wrong!' }
+        format.js
+      end
+    end
+  end
+
+  def create
+    @comment = current_user.comments.build(comment_params)
+    respond_to do |format|
+      if @comment.save
+        format.html do
+          redirect_to @comment.post,
+                      success: 'Comment succesfully created!'
+        end
+        format.js
+      else
+        format.html do
+          redirect_back fallback_location: root_path,
+                        danger: 'Something went wrong!'
+        end
+        format.js
+      end
+    end
+  end
+
+  def destroy
+    unless CommentPolicy.new(current_user, @comment).destroy?
```

Comment:
Being a sucker for simple code and related to my similar annotation in `friend_requests_controller` I must also ask why we used this instead of the simple `Comment` class?